solution/ui/regulations/eregs-vite/src/components/policy-repository/SubjectSelector.vue

```diff
@@ -15,42 +20,140 @@ const props = defineProps({
+            let displayName;
+
+            if (shortNameMatch) {
+                displayName = subject.short_name;
+            } else if (abbreviationMatch) {
+                displayName = subject.abbreviation;
+            } else if (fullNameMatch) {
+                displayName = subject.full_name;
+            }
+
+            displayName =
+                "<span class='match__container'>" +
+                displayName.replace(
+                    new RegExp(filter, "gi"),
+                    (match) => `<span class="match">${match}</span>`
+                ) +
+                "</span>";
+
+            const newSubject = {
+                ...subject,
+                displayName,
+            };
+
+            return [...acc, newSubject];
+        }
+
+        return acc;
+    }, []);
+};
+
+const debouncedFilter = _debounce(getFilteredSubjects, 100);
+
+watch(() => state.filter, debouncedFilter);
+
 const subjectClick = (event) => {
     const subjects = $route?.query?.subjects ?? [];
     const subjectsArray = _isArray(subjects) ? subjects : [subjects];
-    const subjectToAdd = event.target.dataset.id;
+    const subjectToAdd = event.currentTarget.dataset.id;
 
     if (subjectsArray.includes(subjectToAdd)) return;
 
     $router.push({
         name: "policy-repository",
         query: {
             ...$route.query,
-            subjects: [...subjectsArray, event.target.dataset.id],
+            subjects: [...subjectsArray, event.currentTarget.dataset.id],
         },
     });
 };
+
+const filterResetClasses = computed(() => ({
+    "subjects__filter-reset": true,
+    "subjects__filter-reset--hidden": !state.filter,
+}));
+
+const filterResetClick = () => {
+    state.filter = "";
+};
 </script>
 
 <template>
     <div class="subjects__select-container">
         <h3>By Subject</h3>
-        <ul tabindex="-1" class="subjects__list">
-            <li
-                v-for="subject in policyDocSubjects.results"
-                :key="subject.id"
-                class="subjects__li sidebar__li"
-            >
-                <button
-                    :data-name="getSubjectName(subject)"
-                    :data-id="subject.id"
-                    :data-testid="`add-subject-${subject.id}`"
-                    :title="subject.full_name"
-                    @click="subjectClick"
-                >
-                    {{ getSubjectName(subject) }}
-                </button>
-            </li>
-        </ul>
+        <div class="subjects__list-container">
+            <template v-if="props.policyDocSubjects.loading">
+                <div class="subjects__loading">
+                    <SimpleSpinner />
+                </div>
+            </template>
+            <template v-else>
+                <form>
+                    <label for="subjectReduce">Filter the subject list</label>
+                    <input id="subjectReduce" v-model="state.filter" type="text" />
```

Comment:
another win for Vue's `v-model`.  Two-way data binding means we don't need to explicitly add an event handler on the input to update local state.